patches/0004-Use-crypto-backends.patch

```diff
@@ -379,19 +380,164 @@ index 33942467784ad3..0282ffa9fa23c8 100644
  	"crypto/internal/cryptotest"
  	fipsaes "crypto/internal/fips140/aes"
  	"encoding/hex"
+diff --git a/src/crypto/cipher/gcm.go b/src/crypto/cipher/gcm.go
+index 5580f96d55a0fb..8d048c95ad0512 100644
+--- a/src/crypto/cipher/gcm.go
++++ b/src/crypto/cipher/gcm.go
+@@ -93,6 +93,15 @@ func newGCM(cipher Block, nonceSize, tagSize int) (AEAD, error) {
+ func NewGCMWithRandomNonce(cipher Block) (AEAD, error) {
+ 	c, ok := cipher.(*aes.Block)
+ 	if !ok {
++		if cipher, ok := cipher.(gcmAble); ok {
++			g, err := cipher.NewGCM(gcmStandardNonceSize, gcmTagSize)
++			if err != nil {
++				return nil, err
++			}
++			if _, ok := g.(randomNonceSealer); ok {
++				return gcmWithRandomNonce{g}, nil
++			}
++		}
+ 		return nil, errors.New("cipher: NewGCMWithRandomNonce requires aes.Block")
+ 	}
+ 	g, err := gcm.New(c, gcmStandardNonceSize, gcmTagSize)
+@@ -103,7 +112,7 @@ func NewGCMWithRandomNonce(cipher Block) (AEAD, error) {
+ }
+ 
+ type gcmWithRandomNonce struct {
+-	*gcm.GCM
++	aead AEAD
+ }
+ 
+ func (g gcmWithRandomNonce) NonceSize() int {
+@@ -157,7 +166,14 @@ func (g gcmWithRandomNonce) Seal(dst, nonce, plaintext, additionalData []byte) [
+ 		plaintext = ciphertext[:len(plaintext)]
+ 	}
+ 
+-	gcm.SealWithRandomNonce(g.GCM, nonce, ciphertext, plaintext, additionalData)
++	if gc, ok := g.aead.(*gcm.GCM); ok {
++		gcm.SealWithRandomNonce(gc, nonce, ciphertext, plaintext, additionalData)
++	} else if gc, ok := g.aead.(randomNonceSealer); ok {
```

Comment:
`gc` seems like an odd name? `SealWithRandomNonce` calls it `g`, and `gcm` would be another short name I'd expect, but not `gc`. I see that those are both already used, but I think we can do better by being more specific, maybe `aeadGCM`?